src/Helmet.jsx

```diff
@@ -265,6 +272,8 @@ const handleClientStateChange = (newState) => {
     updateTags(TAG_NAMES.META, metaTags);
     updateTags(TAG_NAMES.BASE, baseTag);
 
+    onChangeClientState(newState);
```

Comment:
I actually like how you did it.  No need to separate them out.  Whatever state coming in, should go unfiltered out to the user.